File: shared/types/payload.ts
export enum ActivitySource {
  HEVY = 'hevy',
  KEISER = 'keiser',
  TEST = 'test'
}

export interface ActivityPayload {
  source: ActivitySource;
  userId: string;
  timestamp: string; // ISO 8601
  originalPayload: any; // The raw webhook/SDK data
  metadata?: Record<string, string>; // Optional traces
}

File: shared/types/payload.go
package shared

type ActivitySource string

const (
	SourceHevy   ActivitySource = "hevy"
	SourceKeiser ActivitySource = "keiser"
	SourceTest   ActivitySource = "test"
)

type ActivityPayload struct {
	Source          ActivitySource    `json:"source"`
	UserId          string            `json:"userId"`
	Timestamp       string            `json:"timestamp"` // ISO 8601
	OriginalPayload interface{}       `json:"originalPayload"`
	Metadata        map[string]string `json:"metadata,omitempty"`
}
